lib/irb.rb

```diff
@@ -1283,24 +1286,52 @@ def each_top_level_statement
       end
     end
 
+    COMMAND_FLAG_REGEXP = /(?<cmd_flag>-[a-zA-Z]+( +\S+)*)/
+    COMMAND_ARG_REGEXP = /(?<cmd_arg>[^-]\S*)/
```

Comment:
`[^-]` matches to space. I think it should be `[^- ]`
```ruby
COMMAND_WITH_ARGS_REGEXP.match?('foo -') #=> false
COMMAND_WITH_ARGS_REGEXP.match?('foo  -') #=> true
# Latter one matches because " -" matches to /[^-]\S*/
```